### PR TITLE
Multiparty update

### PR DESCRIPTION
**Pull request by mojtaba-komeili**
**Patch description**
* Added more details to the crowdsourcing task.
* Updating parts of the task with the new LIGHT data model format.

**Testing steps**
Ran it locally and connecting locally with three different usernames, the chat interface works fine.

Screenshot 2023-03-29 at 2 10 34 PM

**Notes**
Some extra files across the light's main functionalities need to be touched for this to work. For example the graph building and one room builder. The changes seemed reasonable in the light of the new data model updates.

### CODE REVIEW

**Comment by JackUrb** [comment]
Merged, I hope the conflicts aren't too bad

File: crowdsourcing/dialogues/multi_party_chat/webapp/src/utils.js [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    'see', 'shan', 'she', 'should', 'shouldn', 'so', 'some',
+    'such', 't', 'than', 'that', 'the', 'their', 'theirs', 'them',
+    'themselves', 'then', 'there', 'these', 'they', 'this',
+    'those', 'through', 'to', 'too', 'under', 'until', 'up', 've',
+    'very', 'want', 'was', 'wasn', 'we', 'were', 'weren', 'what',
+    'when', 'where', 'which', 'while', 'who', 'whom', 'why', 'will',
+    'with', 'won', 'wouldn', 'y', 'you', 'your', 'yours',
+    'yourself', 'yourselves'];
+
+const OFFENSIVE_WORDS = ['2g1c', '2 girls 1 cup', 'acrotomophilia',
```
Comment (JackUrb) [inline crowdsourcing/dialogues/multi_party_chat/webapp/src/utils.js:26]: Have we just been storing and re-copying this list? I would've expected it to be somewhere backend. (not to be changed right now though!)

File: light/graph/builders/one_room_builder.py [hunk 2]
```diff
@@ -22,15 +23,19 @@
...
 from dataclasses import dataclass, field
 from omegaconf import MISSING, DictConfig  # type: ignore
 from typing import Dict, Any, List, Optional, Tuple, TYPE_CHECKING
 
+from light.registry.model_pool import  ModelTypeName
+
+from light.registry.model_pool import  ModelTypeName
+
+from light.registry.model_pool import  ModelTypeName
+
```
Comment (JackUrb) [inline light/graph/builders/one_room_builder.py:38]: Hm probably merge weirdness.

File: light/graph/builders/one_room_builder.py [hunk 3]
```diff
@@ -4,6 +4,7 @@
 # This source code is licensed under the MIT license found in the
 # LICENSE file in the root directory of this source tree.
 
+from parlai.core.params import ParlaiParser
```
Comment (JackUrb) [inline light/graph/builders/one_room_builder.py:7]: I don't know that this import is needed

File: light/registry/models/starspace_model.py [hunk 4]
```diff
@@ -60,6 +61,7 @@ def get_model(self, overrides: Optional[Dict[str, Any]] = None) -> Agent:
                 self.config.resource_path, "object_full_cands.txt"
             )
         else:
+            breakpoint()
```
Comment (JackUrb) [inline light/registry/models/starspace_model.py:64]: nit

File: crowdsourcing/dialogues/multi_party_chat/worlds.py [hunk 5]
```diff
@@ -1,83 +1,294 @@
...
+    AgentShutdownError,
+)
+from mephisto.abstractions._subcomponents.agent_state import AgentState
+
+import constants
+from acceptability import MultiPartyChatChecker
 
 
 SHORT_FORCED_TIMEOUT_TIME = 0.0001
+mephisto_db = LocalMephistoDB()
```
Comment (JackUrb) [inline crowdsourcing/dialogues/multi_party_chat/worlds.py:29]: Needed?

**Review by JackUrb — APPROVED** [review]
Generally looks good, just a few small nits and then feel free to merge!

File: light/graph/builders/one_room_builder.py (hunk 2 above)
Comment (mojtaba-komeili) [reply light/graph/builders/one_room_builder.py:38]: Yeah, cleaned up the imports here a bit.

File: crowdsourcing/dialogues/multi_party_chat/worlds.py (hunk 5 above)
Comment (mojtaba-komeili) [reply crowdsourcing/dialogues/multi_party_chat/worlds.py:29]: Probably not. Removed it!

File: crowdsourcing/dialogues/multi_party_chat/webapp/src/utils.js (hunk 1 above)
Comment (mojtaba-komeili) [reply crowdsourcing/dialogues/multi_party_chat/webapp/src/utils.js:26]: I agree, this was implemented a bit hacky here. I will merge it like this for now, but we can have it pulling from a static page in the future (maybe one of our own Github files).

**Pull request merged**